geomesa-datastore-bundle/geomesa-datastore-processors/src/main/scala/org/geomesa/nifi/datastore/processor/AvroIngestProcessor.scala

```diff
@@ -11,30 +11,94 @@ package org.geomesa.nifi.datastore.processor
 
 import java.io.InputStream
 
+import org.apache.nifi.components.PropertyDescriptor
 import org.apache.nifi.flowfile.FlowFile
 import org.apache.nifi.processor._
 import org.apache.nifi.processor.io.InputStreamCallback
+import org.geomesa.nifi.datastore.processor.AbstractGeoIngestProcessor.checkCompatibleSchema
+import org.geomesa.nifi.datastore.processor.AvroIngestProcessor.{AvroMatchMode, buildWriter}
 import org.geotools.data._
+import org.geotools.feature.simple.SimpleFeatureBuilder
+import org.geotools.util.Converters
 import org.locationtech.geomesa.features.avro.AvroDataFileReader
 import org.locationtech.geomesa.utils.geotools._
-import org.opengis.feature.simple.SimpleFeatureType
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
 
+import scala.util.{Failure, Success}
 import scala.util.control.NonFatal
 
+object AvroIngestProcessor {
+  val AvroMatchMode: PropertyDescriptor =
+    new PropertyDescriptor.Builder()
+      .name("Avro SFT match mode")
+      .description("Determines how Avro SFT mismatches are handled")
+      .required(false)
+      .defaultValue("strict")
+      .allowableValues("strict", "lenient")
+      .build()
+
+  def buildWriter(sft1: SimpleFeatureType,
+                  sft2: SimpleFeatureType,
+                  matchMode: String): (FeatureWriter[SimpleFeatureType, SimpleFeature], SimpleFeature) => Unit = {
+    checkCompatibleSchema(sft1, sft2) match {
+      case Success(true) =>
+        // Schemas match, so use a regular writer.
+        // TODO:  Discuss if the useProvidedFid should be true!
+        (fw: FeatureWriter[SimpleFeatureType, SimpleFeature], sf: SimpleFeature) =>
+          FeatureUtils.write(fw, sf, true)
+      case Failure(error) =>
+        if (matchMode == "lenient") {
+          val sfConverter = convert(sft1, sft2)
+          (fw: FeatureWriter[SimpleFeatureType, SimpleFeature], sf: SimpleFeature) => {
+            val sfToWrite = sfConverter(sf)
+            FeatureUtils.write(fw, sfToWrite, true)
+          }
+        } else {
+          throw error
+        }
+    }
+  }
+
+  import scala.collection.JavaConverters._
+  // Creates an adapter from one SFT to another
+  def convert(in: SimpleFeatureType, out: SimpleFeatureType): SimpleFeature => SimpleFeature = {
+    val out_geometry_ln = out.getGeometryDescriptor.getLocalName
```

Comment:
[picky] usually we use camel case naming conventions for variables

Reply:
As a question....  since we are doing this in NiFi, I think we should enable this looser ingest matching in the command line tools.  (Hence https://geomesa.atlassian.net/browse/GEOMESA-2905)

If that's the case, I wonder if this function should be pushed up to GeoMesa-utils and used there?  Thoughts?

(If so, I'll bang out GM-2905 with this factored up, and we can have this PR use the function from there.)

Reply:
have you tried the CLI tools? i think it should already work, since it creates a converter, and the order of attributes in a converter definition doesn't matter